src/easier68k/core/opcodes/move.py

```diff
@@ -1,54 +1,104 @@
 """
->>> str(Move.from_str('MOVE.B', '-(A0), D1'))
+>>> str(from_str('MOVE.B', '-(A0), D1'))
 'Move command: Size B, src Mode: 4, Data: 0, dest Mode: 0, Data: 1'
 
->>> str(Move.from_str('MOVE.L', 'D3, (A0)'))
+>>> str(from_str('MOVE.L', 'D3, (A0)'))
 'Move command: Size L, src Mode: 0, Data: 3, dest Mode: 2, Data: 0'
 
->>> Move.from_str('MOVE.W', 'D3, A3')[1]
+>>> from_str('MOVE.W', 'D3, A3')[1]
 [('Invalid addressing mode', 'ERROR')]
 """
 from ...core.enum.ea_mode import EAMode
 from ...core.enum.op_size import MoveSize
 from ...core.enum.ea_mode_bin import EAModeBinary
 from ...simulator.m68k import M68K
 from ...core.opcodes.opcode import Opcode
+from ..util.parsing import from_str_util
 
+# Allowed values: nothing, or some combination of B, W, and L (for byte, word, and long)
+# For example, MOVE would have 'BWL' because it can operate on any size of data, while MOVEA would have 'WL' because
+# it can't operate on byte-sized data
+allowed_sizes = 'BWL'
 
-class Move(Opcode):
-    # Allowed values: nothing, or some combination of B, W, and L (for byte, word, and long)
-    # For example, MOVE would have 'BWL' because it can operate on any size of data, while MOVEA would have 'WL' because
-    # it can't operate on byte-sized data
-    allowed_sizes = 'BWL'
+def is_valid(command: str, parameters: str) -> (bool, list):
+    """
+    Tests whether the given command is valid
 
-    @classmethod
-    def from_str(cls, command: str, parameters: str):
-        """
-        Parses a MOVE command from memory.
+    >>> is_valid('MOVE.B', 'D0, D1')[0]
+    True
 
-        :param command: The command itself (e.g. 'MOVE.B', 'LEA', etc.)
-        :param parameters: The parameters after the command (such as the source and destination of a move)
-        """
-        valid, issues = Move.is_valid(command, parameters)
-        if not valid:
-            return None, issues
-        # We can forego asserts in here because we've now confirmed this is valid assembly code
+    >>> is_valid('MOVE.W', 'D0')[0]
+    False
 
+    >>> is_valid('MOVE.G', 'D0, D1')[0]
+    False
+
+    >>> is_valid('MOVE.L', 'D0, A2')[0]
+    False
+
+    >>> is_valid('MOV.L', 'D0, D1')[0]
+    False
+
+    >>> is_valid('MOVE.', 'D0, D1')[0]
+    False
+
+    :param command: The command itself (e.g. 'MOVE.B', 'LEA', etc.)
+    :param parameters: The parameters after the command (such as the source and destination of a move)
+    :return: Whether the given command is valid and a list of issues/warnings encountered
+    """
+    issues = []
+    try:
         parts = command.split('.')  # Split the command by period to get the size of the command
```

Comment:
Couldn't this be replaced with the new `from_str_util` method?

Reply:
not without modification. the util method returns the size and params, but not the parts themselves that are compared by this function. Maybe I could modify the method to return all three and reuse some code.